### PR TITLE
[BugFix][Core] Fix BlockManagerV2 when Encoder Input is None

### LINKED ISSUE #9099: [Bug]: Llama-3.2-11B-Vision-Instruct which is an encoder-decoder model fails with BlockManager V2
### Your current environment

The tests fail on all envs

### Model Input Dumps

_No response_

### 🐛 Describe the bug

The following 2 tests fail with BlockManagerV2 
- tests/models/encoder_decoder/vision_language/test_mllama.py::test_models[5-128-bfloat16-sizes0-meta-llama/Llama-3.2-11B-Vision-Instruct] 
- tests/models/encoder_decoder/vision_language/test_mllama.py::test_models[5-128-bfloat16-sizes4-meta-llama/Llama-3.2-11B-Vision-Instruct] 

The test fail with the following error

```
Traceback (most recent call last):
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/utils.py", line 443, in wrapper
    f(*args, **kwargs)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/models/encoder_decoder/vision_language/test_mllama.py", line 249, in test_models
    run_test(
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/models/encoder_decoder/vision_language/test_mllama.py", line 138, in run_test
    _run_test(
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/models/encoder_decoder/vision_language/test_mllama.py", line 187, in _run_test
    vllm_outputs_per_image = [
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/models/encoder_decoder/vision_language/test_mllama.py", line 188, in <listcomp>
    vllm_model.generate_greedy_logprobs(prompts,
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/conftest.py", line 768, in generate_greedy_logprobs
    return self.generate_w_logprobs(prompts,
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/tests/conftest.py", line 708, in generate_w_logprobs
    req_outputs = self.model.generate(inputs,
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/utils.py", line 1051, in inner
    return fn(*args, **kwargs)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/entrypoints/llm.py", line 391, in generate
    outputs = self._run_engine(use_tqdm=use_tqdm)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/entrypoints/llm.py", line 899, in _run_engine
    step_outputs = self.llm_engine.step()
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/engine/llm_engine.py", line 1357, in step
    ) = self.scheduler[virtual_engine].schedule()
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/scheduler.py", line 1218, in schedule
    scheduler_outputs: SchedulerOutputs = self._schedule()
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/scheduler.py", line 1178, in _schedule
    return self._schedule_default()
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/scheduler.py", line 1013, in _schedule_default
    prefills = self._schedule_prefills(budget,
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/scheduler.py", line 949, in _schedule_prefills
    self._allocate_and_set_running(seq_group)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/scheduler.py", line 1411, in _allocate_and_set_running
    self.block_manager.allocate(seq_group)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/block_manager_v2.py", line 198, in allocate
    block_table = self._allocate_sequence(encoder_seq)
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/block_manager_v2.py", line 154, in _allocate_sequence
    block_table.allocate(seq.get_token_ids())
  File "/home/jovyan/sroy-enc-dec-blk-mgr-fix/vllm/core/block/block_table.py", line 95, in allocate
    assert token_ids
AssertionError

```

Looking into it. This is currently blocking #9084  and #8704.

cc: @afeldman-nm 

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by sroy745**
This PR addresses an issue encountered with certain encoder-decoder models, e.g. Llama-3.2-11B-Vision-Instruct, where encoderInput is set to None. This situation arises in the following test cases:

tests/models/encoder_decoder/vision_language/test_mllama.py::test_models[5-128-bfloat16-sizes0-meta-llama/Llama-3.2-11B-Vision-Instruct]
tests/models/encoder_decoder/vision_language/test_mllama.py::test_models[5-128-bfloat16-sizes4-meta-llama/Llama-3.2-11B-Vision-Instruct]

Currently, these tests function correctly with BlockManagerV1 because, when encoder_input is None during the allocation of cross_block_table, an empty BlockTable is [created without any blocks](https://sourcegraph.com/github.com/vllm-project/vllm@0cc566ca8fd2d21a94f3a8e48bf5c5b60d42b59f/-/blob/vllm/core/block_manager_v1.py?L367). However, this behavior does not hold for BlockManagerV2, where the addition of a BlockTable and attempts to allocate blocks for an empty or None encoder_input lead to assertion failures.

We make the  following changes in this PR:
1. For cases where encoder_input is None, we create an empty BlockTable, similar to the behavior in BlockManagerV1, but without adding any blocks to the BlockTable.
2. We have removed two assertions in BlockManagerV2 that pertain to free blocks and the calculation of physical_block_ids. These assertions check whether the BlockTable is empty, which is valid given the use case. The removal of these assertions will not impact the functionality of the BlockManager.
3. With these changes, we are re-enabling BlockManagerV2 for encoder-decoder models.

FIX #9099
FIX #9084

### CODE REVIEW

**Comment by sroy745** [comment]
@comaniac / @heheda12345 PTAL when you get a chance

File: vllm/core/block_manager_v2.py [hunk 1]
```diff
@@ -151,7 +151,9 @@ def _allocate_sequence(self, seq: Sequence) -> BlockTable:
             block_allocator=self.block_allocator,
             max_block_sliding_window=self.max_block_sliding_window,
         )
-        block_table.allocate(seq.get_token_ids())
+        if (len(seq.get_token_ids()) > 0):
```
Comment (comaniac) [inline vllm/core/block_manager_v2.py:154]:
```suggestion
        if seq.get_token_ids():
```

**Review by comaniac — APPROVED** [review]
LGTM

**Comment by heheda12345** [comment]
LGTM too.

**Pull request merged**